sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -59,6 +62,11 @@ public boolean isEnabled() {
     final RequestDetailsResolver requestDetailsResolver = new RequestDetailsResolver(options);
     transport = transportFactory.create(options, requestDetailsResolver.resolve());
 
+    metricsAggregator =
+        options.isEnableMetrics()
+            ? new MetricsAggregator(options, this)
+            : NoopMetricsAggregator.getInstance();
```

Comment:
in theory, there can be multiple Clients, so multiple MetricsAggregators. We have to set it only once in the options or through a Singleton.

Reply:
Yes, exactly! I think it's fine this way, because:
* per default there's one client being re-used across hubs, so it would also be only one `MetricsAggregator` instance
* the only use-case I could think of having different clients would be having different DSNs within a single app, in this case it should be fine as well, as we would aggregate per DSN
* it's the same way in [our Python SDK](https://github.com/getsentry/sentry-python/blob/2eeb8c50a0fe987cf70ef254ea0d63bf422a1899/sentry_sdk/client.py#L261) 

But I'm definitely not an expert on this, @adinauer maybe has some insights when people decide to have multiple `SentryClient` instances.

Reply:
I'm not sure how this should work if there's multiple clients. After merging Hubs and Scopes there'll be multiple scopes and setting a different client should in theory be quite easy. I can try and find someone who can answer this question.

Reply:
Each client has its own aggregator, we stick to the same architecture as in Python.